sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/AbstractMethodParameterLookup.cs

```diff
@@ -72,11 +72,16 @@ public bool TryGetParameterSymbol(TArgumentSyntax argument, out IParameterSymbol
 
         public bool TryGetSymbolParameter(IParameterSymbol parameter, out TArgumentSyntax argument)
         {
-            foreach(var Pair in GetAllArgumentParameterMappings())
+            return TryGetArgumentSyntax(parameter.Name, out argument);
+        }
+
+        public bool TryGetArgumentSyntax(string parameterName, out TArgumentSyntax argument)
+        {
+            foreach (var pair in GetAllArgumentParameterMappings())
             {
-                if(Pair.Symbol == parameter)
+                if (parameterName.Equals(pair.Symbol.Name))
```

Comment:
string.Equals is equivalent to == for strings, except the NullReference risk. I'd stay with original == test in case someone calls this with null argument.